docs/glossary.md

```diff
@@ -181,6 +181,16 @@ function `root()`. Note that our definition of "root commit" is different from
 Git's; Git's "root commits" are the first commit(s) in the repository, i.e. the
 commits `jj log -r 'root()+'` will show.
 
+## same-change rule
+
+The same-change rule is a pragmatic hack on the conflict algebra, which allows
+us to consider a conflict as automatically resolved when two sides are the same.
+This is similar to Git and Mercurial (hg) and not like Darcs and its
+descendants which would consider it a conflict. It also has some unfortunate
+consequences like when you rebase a commit to upstream and the upstream rebased
+it  to where it was before, you may lose changes (any that overlapped with
+upstream's changes).
```

Comment:
I felt a little hesitant about including this in the glossary before when I saw this PR because it doesn't seem like something users would need to know. After thinking a bit more about it, I feel like docs/technical/conflicts.md is a better place for this. What do you think?

Reply:
> After thinking a bit more about it, I feel like docs/technical/conflicts.md is a better place for this. What do you think?

Isn't the Glossary the place to make references to though? I think it deserves a outline in the conflicts doc though with a reference to this.

Reply:
I don't think internal terms belong in the glossary. If we end up with many internal terms, I suppose we can create an internal/technical glossary, but I don't think we're at that point yet.

Reply:
Ok, I've moved it although it now needs a major rework.